src/tendril.rs

```diff
@@ -906,7 +1062,11 @@ impl<F> Tendril<F>
     }
 }
 
-impl DerefMut for Tendril<fmt::Bytes> {
+// There's no need to worry about locking on an atomic Tendril, because it makes it unique as
+// soon as you do that.
```

Comment:
It was a fair assumption, the problem was already there before your change.

Nit: what I mind was making `make_owned_with_capacity` call `make_owned` rather than the reverse, so that the latter doesn’t need to call `buf.grow` at all. I’ve made a PR into your PR: https://github.com/servo/tendril/pull/17